src/Compilers/CSharp/Test/Semantic/Utilities/ValueSetTests.cs

```diff
@@ -0,0 +1,663 @@
+            var letters = gea1.Intersect(lez1).Union(gea2.Intersect(lez2));
+            Assert.Equal("['A'..'Z'],['a'..'z']", letters.ToString());
+        }
+
+        [Fact]
+        public void TestDouble_02()
+        {
+            Assert.Equal("-Inf", ForDouble.Related(LessThan, double.MinValue).ToString());
+            var lt = ForDouble.Related(LessThan, 0.0);
+            Assert.Equal(FormattableString.Invariant($"-Inf,[{double.MinValue:G17}..{-double.Epsilon:G17}]"), lt.ToString());
+            var gt = ForDouble.Related(GreaterThan, 0.0);
+            Assert.Equal(FormattableString.Invariant($"Inf,[{double.Epsilon:G17}..{double.MaxValue:G17}]"), gt.ToString());
+            var eq = ForDouble.Related(Equal, 0.0);
+            Assert.Equal("[0..0]", eq.ToString());
+            var none = lt.Complement().Intersect(gt.Complement()).Intersect(eq.Complement());
+            Assert.Equal("NaN", none.ToString());
+            Assert.False(none.IsEmpty);
+        }
+
+        [Fact]
+        public void TestFloat_01()
+        {
+            Assert.Equal("-Inf", ForFloat.Related(LessThan, float.MinValue).ToString());
+            var lt = ForFloat.Related(LessThan, 0.0f);
+            Assert.Equal(FormattableString.Invariant($"-Inf,[{float.MinValue:G9}..{-float.Epsilon:G9}]"), lt.ToString());
+            var gt = ForFloat.Related(GreaterThan, 0.0f);
+            Assert.Equal(FormattableString.Invariant($"Inf,[{float.Epsilon:G9}..{float.MaxValue:G9}]"), gt.ToString());
+            var eq = ForFloat.Related(Equal, 0.0f);
+            Assert.Equal("[0..0]", eq.ToString());
+            var none = lt.Complement().Intersect(gt.Complement()).Intersect(eq.Complement());
+            Assert.Equal("NaN", none.ToString());
+            Assert.False(none.IsEmpty);
+        }
+
+        [Fact]
+        public void TestDouble_03()
+        {
+            Assert.Equal("NaN", ForDouble.Related(Equal, double.NaN).ToString());
+            Assert.Equal("NaN", ForFloat.Related(Equal, float.NaN).ToString());
+            Assert.True(ForDouble.Related(Equal, double.NaN).Any(Equal, double.NaN));
+            Assert.True(ForFloat.Related(Equal, float.NaN).Any(Equal, float.NaN));
+            Assert.Equal("Inf", ForDouble.Related(Equal, double.PositiveInfinity).ToString());
+            Assert.Equal("Inf", ForFloat.Related(Equal, float.PositiveInfinity).ToString());
+            Assert.Equal("-Inf", ForDouble.Related(Equal, double.NegativeInfinity).ToString());
+            Assert.Equal("-Inf", ForFloat.Related(Equal, float.NegativeInfinity).ToString());
+        }
+
+        [Fact]
+        public void TestDouble_04()
+        {
+            var neg = ForDouble.Related(LessThan, 0.0);
+            Assert.True(neg.Any(LessThan, double.MinValue));
+            Assert.False(neg.Any(GreaterThan, double.MaxValue));
+
+            var mi = ForDouble.Related(Equal, double.NegativeInfinity);
+            Assert.True(mi.All(LessThan, 0.0));
+        }
```

Comment:
>} [](start = 8, length = 1)

Consider testing `mi.Any(GreaterThan, 0.0)`.

Reply:
Will do so in next PR.

---
In reply to: [383440855](https://github.com/dotnet/roslyn/pull/41503#discussion_r383440855) [](ancestors = 383440855)